### PR TITLE
Make Paths equatable, add tests

### LINKED ISSUE #1815: Cadence Paths Equality Comparison
### Issue To Be Solved

Currently it is not possible to check Cadence Paths for equality

`path  != (/storage/flowTokenVault as Path)`

results in:

`cannot apply binary operation != to types: "Path", "Path"`

### Suggested Solution

Paths should be comparable.

### PR DESCRIPTION
**Pull request by dreamsmasher**
Closes #1815

## Description

Turns out path equality was implemented but disabled.
______

- [x] Targeted PR against `master` branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/cadence/blob/master/CONTRIBUTING.md#styleguides)
- [x] Updated relevant documentation 
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

### CODE REVIEW

File: runtime/tests/interpreter/interpreter_test.go [hunk 1]
```diff
@@ -1055,6 +1055,25 @@ func TestInterpretEqualOperator(t *testing.T) {
...
+		return /public/foo == /public/foofoo ||
+			  /private/bar == /private/barbar ||
+			  /storage/baz == /storage/bazbaz
+	  }
+
+	  fun testCastedPaths(): Bool {
+		let foo: StoragePath = /storage/foo
+		let bar: PublicPath = /public/foo
+		return (foo as Path) == (bar as Path)
+	  }
```
Comment (dsainati1) [inline runtime/tests/interpreter/interpreter_test.go:1076]: Mixing tabs and spaces can cause formatting to look weird on GitHub, hence these three functions having different indentation than the others. If you find and replace all tabs with 4 spaces in a file, the Go autoformatter should replace those with tabs everywhere except inside the test strings.

**Review by dsainati1 — COMMENTED** [review]
Nice! Glad you were able to find all this, and that it turned out to be so simple! Just one quick formatting request.

**Review by dsainati1 — APPROVED** [review]
Nice!

**Review by turbolent — APPROVED** [review]
Great work! First PR 🎉 

I had no idea it was so simple to implement, didn't realize that `PathValue` already implements `EquatableValue` 👍

**Comment by janezpodhostnik** [comment]
Thank you!

**Pull request merged**